src/Api/PersistentImplementation.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE ImpredicativeTypes          #-}
+{-# LANGUAGE OverloadedStrings           #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TypeOperators               #-}
+
+{-# OPTIONS_GHC -Wall -Werror -fno-warn-orphans #-}
+
+module Api.PersistentImplementation
```

Comment:
You mean "How about `Api.Persistent.STM`?" in the sense that you propose to rename the module? A good point, given that we already know we will have more than one implementation. Normally, I'd even move the implementation to a distant directory called `Implementations`, somewhere close to the top-level, together with an implementation(s) of `ActionM`, etc. to make sure people are aware they are mucking around with implementation details. How about `Api.PersistentImplementation.STM` as a compromise?